### PR TITLE
ContinuousPHP configuration for PDO Oracle driver

### PR DESCRIPTION
**Pull request by morozov**
|      Q       |   A
|------------- | -----------
| Type         | improvement
| BC Break     | no
| Fixed issues | #3294

### CODE REVIEW

**Review by Ocramius — DISMISSED** [review]

**Comment by morozov** [comment]
Filed [bug #77118](https://bugs.php.net/bug.php?id=77118) in PHP and https://github.com/php/php-src/pull/3655.

**Comment by morozov** [comment]
The [`PDO_OCI` build](https://app.continuousphp.com/git-hub/doctrine/dbal/build/e3aefaa9-4901-49bb-afa2-52b68150c9be/logs/0e51499b-d3c5-4ce2-88eb-ce799bc95c49) passed. Currently, ContinuosPHP is configured to run it only against this branch which is done via a UI. Once these changes are merged, I'll enable it for all branches.

File: lib/Doctrine/DBAL/Driver/PDOConnection.php [hunk 1]
```diff
@@ -104,7 +104,11 @@ public function quote($input, $type = ParameterType::STRING)
      */
     public function lastInsertId($name = null)
     {
-        return parent::lastInsertId($name);
+        try {
+            return parent::lastInsertId($name);
+        } catch (\PDOException $exception) {
```
Comment (Majkl578) [inline lib/Doctrine/DBAL/Driver/PDOConnection.php]: Is this covered by a test somewhere?

File: tests/Doctrine/Tests/DBAL/Functional/DataAccessTest.php [hunk 2]
```diff
@@ -5,6 +5,10 @@
 use DateTime;
 use Doctrine\DBAL\Connection;
 use Doctrine\DBAL\Driver\Mysqli\Driver as MySQLiDriver;
+use Doctrine\DBAL\Driver\OCI8\Driver as Oci8Driver;
+use Doctrine\DBAL\Driver\PDOConnection;
+use Doctrine\DBAL\Driver\PDOOracle\Driver as PDOOracleDriver;
+use Doctrine\DBAL\Driver\PDOSqlsrv\Driver as PDOSQLSRVDriver;
```
Comment (Majkl578) [inline tests/Doctrine/Tests/DBAL/Functional/DataAccessTest.php]: Should be `PDOSqlsrvDriver` to follow original name.

File: tests/Doctrine/Tests/DBAL/Functional/WriteTest.php [hunk 3]
```diff
@@ -334,4 +335,25 @@ public function testDeleteWhereIsNull()
...
+     * @return string
+     *
+     * @throws DriverException
+     */
+    private function lastInsertId(?string $name = null)
+    {
+        try {
+            return $this->connection->lastInsertId($name);
+        } catch (DriverException $e) {
+            if ($e->getCode() === 'IM001') {
```
Comment (Majkl578) [inline tests/Doctrine/Tests/DBAL/Functional/WriteTest.php]: This looks fragile, is it possible it may hide possible bugs in future?

**Review by Majkl578 — COMMENTED** [review]
👍 overall, just few little things.

File: lib/Doctrine/DBAL/Driver/PDOConnection.php (hunk 1 above)
Comment (morozov) [reply lib/Doctrine/DBAL/Driver/PDOConnection.php]: Yes, in https://github.com/doctrine/dbal/pull/3339#discussion_r238083147.

File: tests/Doctrine/Tests/DBAL/Functional/WriteTest.php (hunk 3 above)
Comment (morozov) [reply tests/Doctrine/Tests/DBAL/Functional/WriteTest.php]: This is how PDO [reports](https://github.com/php/php-src/blob/1ad08256f349fa513157437abc4feb245cce03fc/ext/pdo/pdo_sqlstate.c#L283) that it doesn't support a certain feature. I think it's the best we can get from PDO.

**Review by Majkl578 — DISMISSED** [review]

**Review by Majkl578 — APPROVED** [review]

**Pull request merged**